modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -88,11 +88,42 @@ export function getValueDetailsFromItem(item, fieldName) {
 		return null;
 	}
 
+	// TODO: extract function to get same behavior as block 130-143
+	const languageId =
+		Liferay.ThemeDisplay.getLanguageId() ||
+		Liferay.ThemeDisplay.getBCP47LanguageId() ||
+		Liferay.ThemeDisplay.getDefaultLanguageId();
+
+	const rawTextFieldName = `${fieldName}RawText`;
+	const i18nFieldName = `${fieldName}_i18n`;
+
 	let rootPropertyName = fieldName;
 	const valuePath = [];
 	let navigatedValue = item;
 
-	if (Array.isArray(fieldName)) {
+	// Cover richText content value
+	// {name: '<p>...</p>', nameRawText: '...'}
+	if (
+		Object.hasOwn(item, rawTextFieldName) &&
+		Object.values(rawTextFieldName).length
+	) {
+		valuePath.push(fieldName);
+		navigatedValue = navigatedValue[rawTextFieldName];
+	}
+	// Cover _i18n fields with language keys
+	// {name: '', name_i18n: {en_US: ''}}
+	else if (Object.hasOwn(item, i18nFieldName) &&
+			Object.values(i18nFieldName).length) {
+		valuePath.push(fieldName);
+		navigatedValue = navigatedValue[i18nFieldName][languageId];
+	}
+	// Cover fields content value with language keys
+	// {name: {en_US: ''}}
+	else if (item[fieldName] && item[fieldName][languageId]) {
+		valuePath.push(fieldName);
```

Comment:
As we talked, this part covers any field with a "localized" structure (i.e. an object in the form `{ "fieldname" : {"en-US": "fieldValue1", "fr-FR": "fieldValue2" } }` ).
Still we need to unify logic to search the values in the locales as the code block processing `LANG` property is doing (line 130 and below)